packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -72,14 +73,20 @@ class MasterDetailLayout extends SlotStylesMixin(ResizeMixin(ElementMixin(Themab
         position: relative;
       }
 
-      :host(:is([overlay], [stack])[containment='layout']) [part='detail'] {
+      :host(:is([overlay], [stack])[containment='layout']) [part='detail'],
+      :host([overlay][containment='layout']) [part='backdrop'] {
         position: absolute;
       }
 
-      :host(:is([overlay], [stack])[containment='viewport']) [part='detail'] {
+      :host(:is([overlay], [stack])[containment='viewport']) [part='detail'],
+      :host([overlay][containment='viewport']) [part='backdrop'] {
         position: fixed;
       }
 
+      :host([overlay]) [part='backdrop'] {
```

Comment:
IMO we should hide the backdrop (`display: none`) when there are no details or when the component is not in overlay mode:
- Currently it is shown when you force overlay, but there are no details
- It is also visible when the component is not in overlay mode, which can lead to unexpected results when you apply a gap on the layout for example